Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM docker.mirror.hashicorp.services/alpine:3.13.6 as default
```

Comment:
Since this replaces the previous release dockerfile, can you also remove `docker/Release.dockerfile` in this PR?

Reply:
Also a question: Is it a requirement of CRT that the dockerfile be at the repo root? Moving the dockerfile out of the `docker` directory makes several things stated in the [docker/README.md](https://github.com/hashicorp/boundary/blob/main/docker/README.md#boundary-docker-image) incorrect.

If the dockerfile must be here that is ok. I can update that README in a followup PR, but I just want to check before doing so.

Reply:
I think we still want the `Dev.dockerfile`, since the `docker-build-dev` make target uses it, and it is helpful for developers. I suppose that file could be moved into the repo root, and the `docker-build-dev` target could be updated. Alternatively I can do those updates in a separate PR since they are not directly related to the release CRT changes, but would just be some cleanup and consolidation of docker for local dev.

> I'm happy to update the README -- do you still want this under the docker directory, or should we move it under .release/docker since there won't be anything else in this dir?

I think if everything else gets removed from the `docker` directory, moving the README to `.release/docker` would be good

Reply:
I just pushed up my latest changes with the different docker targets defined in the root Dockerfile, and the docker README moved to `.release/docker` and a few edits made there. Let me know what you think- can split this out into another PR if needed.

Reply:
Nice, this all looks good. I think it make sense to keep this all in the same PR